minimint-api/src/db/rocksdb_impl.rs

```diff
@@ -25,7 +25,13 @@ impl Database for rocksdb::OptimisticTransactionDB {
         let prefix = key_prefix.to_vec();
         Box::new(
             self.prefix_iterator(prefix.clone())
-                .map_while(move |res| res.0.starts_with(&prefix).then(|| res))
+                .map_while(move |res| {
+                    res.as_ref()
+                        .unwrap()
+                        .0
+                        .starts_with(&prefix)
+                        .then(|| res.unwrap())
```

Comment:
I think you should avoid both instances of `.unwrap()` here because they will panic if `res` is an `Err`. You might be able to convert the `Result` into an `Option` and then use [`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) for example.

Reply:
I've just amended the commit with a filter to make sure the results are okay